Frameworks/Foundation/NSArray.mm

```diff
@@ -1041,32 +1028,43 @@ - (NSArray*)sortedArrayUsingFunction:(NSInteger (*)(id, id, void*))comparator co
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (NSUInteger)indexOfObjectIdenticalTo:(id)anObject inRange:(NSRange)range {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (range.location + range.length > [self count]) {
+        [NSException raise:NSRangeException
+                    format:@"Range {%d, %d} larger than array of size %d.", range.location, range.length, [self count]];
+    }
+
+    NSUInteger end = NSMaxRange(range);
+    for (NSUInteger i = range.location; i < end; ++i) {
+        if ([self objectAtIndex:i] == anObject) {
+            return i;
+        }
+    }
+
+    return NSNotFound;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (NSUInteger)indexOfObjectWithOptions:(NSEnumerationOptions)opts passingTest:(BOOL (^)(id, NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSIndexSet* fullRange = [NSIndexSet indexSetWithIndexesInRange:NSMakeRange(0, self.count)];
+    return [self indexOfObjectAtIndexes:fullRange options:opts passingTest:predicate];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (NSUInteger)indexOfObjectAtIndexes:(NSIndexSet*)indexSet
                              options:(NSEnumerationOptions)opts
                          passingTest:(BOOL (^)(id, NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSIndexSet* matching = [self indexesOfObjectsAtIndexes:indexSet options:opts passingTest:predicate];
```

Comment:
Optimization:
If you add another level of block here, you can stop enumeration on the first hit instead of continuing to aggregate every index and then discarding the rest of them.